File: python/structures/src/math/eratosthenes.py
def sieve(max_num):
    if max_num < 2:
        return []

    primes = [x for x in range(2, max_num)]

    i = 0

    while i < len(primes) - 1:
        j = i + 1

        while j < len(primes):
            if primes[j] % primes[i] == 0:
                del primes[j]
            else:
                j += 1

        i += 1

    return primes

File: ruby/src/math/eratosthenes.rb
module Eratosthenes
    def Eratosthenes.sieve(up_to)
        if up_to < 2
            []
        end

        primes = (2..up_to).to_a

        i = 0

        # sieve out non-primes
        while i < primes.length - 1
            j = i + 1

            while j < primes.length
                if primes[j] % primes[i] == 0
                    primes.delete_at(j)
                else
                    j += 1
                end
            end

            i += 1
        end

        primes
    end
end
